src/olmo_core/train/callbacks/evaluator_callback.py

```diff
@@ -184,21 +183,20 @@ def __init__(
         if is_distributed():
             sampler = DistributedSampler(
                 self.task,  # type: ignore
-                drop_last=True,
+                drop_last=False,
                 shuffle=False,
                 num_replicas=get_world_size(dp_process_group),
                 rank=get_rank(dp_process_group),
             )
 
-        rank_batch_size_instances = max(0, rank_batch_size // self.task.max_sequence_length)
```

Comment:
This causes a bug. I don't see why we should divide batch size by seq len. Batch size was already number of examples.

Reply:
> Batch size was already number of examples.

Where was it set to number of examples instead of tokens? It should always be set in tokens. This change will cause bugs elsewhere

Reply:
It's derived from Line 230 of this file, `eval_batch_size`. Do you mean this variable should be in number of tokens? It seems to be a semantic change from the old repo, and each eval's max seq length is dependent on the task so we can't set a fixed batch size ...

Reply:
It is a change from the old repo.

> each eval's max seq length is dependent on the task so we can't set a fixed batch size

Well batch size is roughly fixed by number of tokens, not instances. This is more efficient because we can pack more instances from shorter tasks together.

Reply:
Got it. I've reverted it so now `eval_batch_size` takes number of tokens.